apps/api/src/submission/entity/submission.entity.ts

```diff
@@ -19,4 +19,41 @@ export class SubmissionEntity extends BaseEntity {
   @Exclude()
   @Column('varchar', { nullable: false })
   version!: string;
+  @BeforeInsert()
+  beforeInsert() {
+    let { contactInformation, skillInformation, availabilityInformation } = this.payload;
+
+    // Remove specialties if the stream is non clinical
+    if (skillInformation.stream && skillInformation.stream !== streamsById.Nonclinical.id) {
+      this.payload.skillInformation.specialties = [];
+    }
+
+    // Remove Health authorities if the user is not a resident or employed
+    if (
+      [EmploymentTypes.HEALTH_SECTOR_EMPLOYED, EmploymentTypes.HEALTH_SECTORY_RESIDENCY].includes(
```

Comment:
Would it be helpful to abstract this conditions and reuse them in API and Common?

Reply:
I don't think it's quite needed. If we run into this again, then it might make more sense.